packages/upload-client/test/helpers/utils.js

```diff
@@ -1,12 +1,67 @@
-import { Receipt } from '@ucanto/core'
+import { parseLink } from '@ucanto/server'
+import * as Signer from '@ucanto/principal/ed25519'
+import { Receipt, Message } from '@ucanto/core'
+import * as CAR from '@ucanto/transport/car'
 import * as Server from '@ucanto/server'
 import * as HTTP from '@web3-storage/capabilities/http'
 import * as W3sBlobCapabilities from '@web3-storage/capabilities/web3.storage/blob'
 import { W3sBlob } from '@web3-storage/capabilities'
 import { createConcludeInvocation } from '../../../upload-client/src/blob.js'
+import { Assert } from '@web3-storage/content-claims/capability'
 
 export const validateAuthorization = () => ({ ok: {} })
 
+// @ts-ignore Parameter
+export const setupGetReceipt = (contentGenFn) => {
```

Comment:
I would recommend to simplify this to either have a separate process mocking the getReceipt endpoint, or create a http server within this process instead of mocking the actual fetch implementation.

Reply:
in w3up-client we are using a mock server as a different process.
run process https://github.com/w3s-project/w3up/blob/604d300dde17521eea7333b949bb3f6796bd80a2/packages/w3up-client/package.json#L115
see that mock runs on npm run test https://github.com/w3s-project/w3up/blob/604d300dde17521eea7333b949bb3f6796bd80a2/packages/w3up-client/package.json#L109
https://github.com/w3s-project/w3up/blob/604d300dde17521eea7333b949bb3f6796bd80a2/packages/w3up-client/test/helpers/receipts-server.js

This is one way to go there, and if you see upload-client already uses this strategy for bucket-server https://github.com/w3s-project/w3up/blob/604d300dde17521eea7333b949bb3f6796bd80a2/packages/upload-client/package.json#L25 to write to a presigned URL

Reply:
there is an alternative that maybe is more on the direction you are taking that you could also consider.  you can rely on node http Server directly, freeway uses this kind of approach here https://github.com/web3-storage/freeway/blob/main/test/helpers/bucket.js. So, instead of a different process to run a mock server, the server runs in same process as the tests, and its URL and port are used by fetch  to perfrom requests against.

Reply:
thank you! I don't think the first approach would work as I need to load dynamic content into the receipt server. the 2nd approach should work and it does make sense, but I'm wondering if it's worth given the client refactor 🤔